docs/bridge/docs/03-RFQ/15-Bridging/zaps.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Zaps
+---
+
+<!-- Reference Links -->
+[relay]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#relay
+[prove]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#prove
+[dispute]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridge.sol/interface.IFastBridge.html#dispute
+[claim]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#claim
+[cancel]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#cancel
+[proof]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#bridgetxdetails
+[BridgeRequested]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridge.sol/interface.IFastBridge.html#bridgerequested
+[BridgeTransactionV2]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridgeV2.sol/interface.IFastBridgeV2.html#bridgetransactionv2
+[BridgeRelayed]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridge.sol/interface.IFastBridge.html#bridgerelayed
+[BridgeProofProvided]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IFastBridge.sol/interface.IFastBridge.html#bridgeproofprovided
+[Cancel Delay]: https://vercel-rfq-docs.vercel.app/contracts/FastBridgeV2.sol/contract.FastBridgeV2.html#refund_delay
+[Multicall]: https://vercel-rfq-docs.vercel.app/contracts/interfaces/IMulticallTarget.sol/interface.IMulticallTarget.html
+
+[Quoter API]: /docs/RFQ/Quoting/Quoter%20API/
+[Dispute Period]: /docs/RFQ/Security/#dispute-period
+[Quoting]: /docs/RFQ/Quoting
+[Bridging]: /docs/RFQ/Bridging
+[Relaying]: /docs/RFQ/Relaying
+[Proving]: /docs/RFQ/Proving
+[Claiming]: /docs/RFQ/Claiming
+[Canceling]: /docs/RFQ/Canceling
+[Security]: /docs/RFQ/Security
+
+[User]: /docs/RFQ/#entities
+[Quoter]: /docs/RFQ/#entities
+[Prover]: /docs/RFQ/#entities
+[Relayer]: /docs/RFQ/#entities
+[Guard]: /docs/RFQ/#entities
+[Canceler]: /docs/RFQ/#entities
+
+# WIP
```

Comment:
Things to add here (again, focusing on the user/integration perspective):

- Setting non-zero `ZapNative` to request additional gas on top of the bridged token.
  - Can be used with recipient EOA.
  - Can not be used with `0xEEEE` destination token.
- Utilising non-empty `ZapData` to request a zap action to be performed using the bridged tokens.
  - Can only be used with smart-contract recipients that follow `IZapRecipient` interface.
  - Highlight that if the zap action fails, the whole transaction gets reverted (aka no stuck funds in the recipient contract). Can be again attempted for relay by same/different Relayer; or cancelled on the origin chain if no one successfully relays.
  - ZapData should be encoded according to the documentation of that specific recipient contract.
    - That will typically use the user address to act on behalf of, and the amount of tokens to use.
    - When a quote is requested, ZapData must be encoded using `originAmount` as amount of tokens to use on the destination chain.
    - After the quote is received, that quoted amount must be used to regenerate ZapData.
- Can use non-zero ZapNative with non-empty ZapData, if the smart contract recipient requires both a token + native gas transfer for the zap action.